packages/plugin-bootstrap/__tests__/actions.test.ts

```diff
@@ -0,0 +1,1442 @@
+import { describe, expect, it, vi, beforeEach, afterEach } from 'vitest';
+import { replyAction } from '../src/actions/reply';
+import { followRoomAction } from '../src/actions/followRoom';
+import { ignoreAction } from '../src/actions/ignore';
+import { muteRoomAction } from '../src/actions/muteRoom';
+import { unmuteRoomAction } from '../src/actions/unmuteRoom';
+import { choiceAction } from '../src/actions/choice';
+import { sendMessageAction } from '../src/actions/sendMessage';
+import { updateEntityAction } from '../src/actions/updateEntity';
+import { noneAction } from '../src/actions/none';
+import updateRoleAction from '../src/actions/roles';
+import updateSettingsAction from '../src/actions/settings';
+import { unfollowRoomAction } from '../src/actions/unfollowRoom';
+import {
+  IAgentRuntime,
+  logger,
+  Entity,
+  ModelType,
+  Memory,
+  State,
+  UUID,
+  Content,
+  ChannelType,
+  HandlerCallback,
+  EventType,
+} from '@elizaos/core';
+import { createMockRuntime, createMockMemory, createMockState, MockRuntime } from './test-utils';
+
+// Additional mock setup for actions testing
+function setupActionTest() {
+  // Create a message from a user directed at our agent
+  const mockMessage: Partial<Memory> = createMockMemory({
+    id: 'test-message-id' as UUID,
+    roomId: 'test-room-id' as UUID,
+    entityId: 'test-entity-id' as UUID,
+    agentId: 'test-agent-id' as UUID,
+    content: {
+      text: 'Hello agent follow this room', // Added "follow" to pass keyword check
+      channelType: ChannelType.GROUP,
+    },
+  });
+
+  // Create a standardized state object
+  const mockState: Partial<State> = createMockState({
+    values: {
+      agentName: 'Test Agent',
+      recentMessages: 'User: Hello agent\nAgent: Hi there!',
+      actionNames:
+        'REPLY, FOLLOW_ROOM, UNFOLLOW_ROOM, IGNORE, MUTE_ROOM, UNMUTE_ROOM, CHOOSE_OPTION, SEND_MESSAGE, UPDATE_ENTITY, NONE',
+      choices: 'Option A|Option B|Option C',
+      choicePrefix: 'Choose option:',
+      choiceDelimiter: '|',
+    },
+    data: {
+      room: {
+        id: 'test-room-id',
+        type: 'group',
+        worldId: 'test-world-id',
+        serverId: 'test-server-id',
+        name: 'Test Room',
+      },
+      currentParticipantState: 'ACTIVE',
+    },
+  });
+
+  // Create a mock runtime with the necessary methods for action testing
+  const mockRuntime = createMockRuntime({
+    // Override default mocks with action-specific implementations
+    useModel: vi.fn().mockImplementation((modelType, params) => {
+      if (modelType === ModelType.OBJECT_LARGE) {
+        return Promise.resolve({
+          thought: 'I should respond in a friendly way',
+          message: 'Hello there! How can I help you today?',
+        });
```

Comment:
We can probably move all this test set up to another file that we reuse across test suites